pkg/cluster/backup_manager.go

```diff
@@ -128,22 +127,47 @@ func (bm *backupManager) runSidecar() error {
 }
 
 func (bm *backupManager) createSidecarDeployment() error {
-	cl, c := bm.cluster, bm.config
+	d := bm.makeSidecarDeployment()
+	_, err := bm.config.KubeCli.AppsV1beta1().Deployments(bm.cluster.Metadata.Namespace).Create(d)
+	return err
+}
 
+func (bm *backupManager) updateSidecar(cl *spec.Cluster) error {
+	// change local structs
+	bm.cluster = cl
+	var err error
+	bm.s, err = bm.setupStorage()
+	if err != nil {
+		return err
+	}
+	// change k8s objects
```

Comment:
probably we should add a func to handle k8s deployment updates. this is a code duplication.